### PR TITLE
[doc] add lazy init tutorial

### PR DESCRIPTION
**Pull request by ver217**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

Add tutorial about lazy initialization for both English and Chinese.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: docs/source/en/features/lazy_init.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+You may use lazy initialization when using Gemini, tensor parallelism, pipeline parallelism, and auto-parallelism. In other cases, you may not need to use lazy initialization.
+
+Gemini is compatible with lazy initialization. You can use them together directly.
+
+<!--- doc-test-ignore-start -->
+```python
+from colossalai.booster import Booster
+from colossalai.booster.plugin import GeminiPlugin
+from colossalai.lazy import LazyInitContext
+
```
Comment (FrankLeeeee) [inline docs/source/en/features/lazy_init.md:32]: All the code is ignored, should we test it?

Comment (ver217) [reply docs/source/en/features/lazy_init.md:32]: This would be tested in unit test of gemini plugin. We can also test it here. I will update.

Comment (ver217) [reply docs/source/en/features/lazy_init.md]: added

File: docs/source/en/features/lazy_init.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Lazy initialization
+
+Author: Hongxin Liu
+
+**Prerequisite**
+- [Booster API](../basics/boolster_api.md)
+- [Booster Plugins](../basics/boolster_plugins.md)
+- [Booster Checkpoint](../basics/boolster_checkpoint.md)
+
```
Comment (FrankLeeeee) [inline docs/source/en/features/lazy_init.md:9]: `boolster` is spelled wrongly.

Comment (ver217) [reply docs/source/en/features/lazy_init.md]: fixed

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**